whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java

```diff
@@ -0,0 +1,136 @@
+package net.ripe.db.whois.api.nrtm4;
+
+import com.google.common.net.HttpHeaders;
+import net.ripe.db.nrtm4.dao.DeltaSourceAwareFileRepository;
+import net.ripe.db.nrtm4.dao.NotificationFileSourceAwareDao;
+import net.ripe.db.nrtm4.dao.SnapshotSourceAwareFileRepository;
+import net.ripe.db.nrtm4.dao.SourceRepository;
+import net.ripe.db.nrtm4.domain.NrtmDocumentType;
+import net.ripe.db.nrtm4.domain.NrtmSourceModel;
+import net.ripe.db.nrtm4.util.NrtmFileUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.ByteArrayInputStream;
+
+@Component
+@Path("/")
+public class NrtmClientService {
+
+    public static final String SOURCE_LINK_PAGE = "<html><header><title>NRTM Version 4</title></header><body>%s<body></html>";
+    private final SnapshotSourceAwareFileRepository snapshotSourceAwareFileRepository;
+    private final DeltaSourceAwareFileRepository deltaSourceAwareFileRepository;
+    private final NotificationFileSourceAwareDao notificationFileSourceAwareDao;
+    private final SourceRepository sourceRepository;
+    final String nrtmUrl;
+
+    @Autowired
+    public NrtmClientService(@Value("${nrtm.baseUrl}") final String nrtmUrl,
```

Comment:
Should this property be mandatory? We need a way to turn NRTMv4 off in certain environments.

Reply:
I will make it optional